mempool/mempool.go

```diff
@@ -22,6 +22,14 @@ import (
 	"github.com/tendermint/tendermint/types"
 )
 
+// PreCheckFunc is an optional filter to determine if a transaction should be
+// rejected. Invoked before CheckTx.
+type PreCheckFunc func(types.Tx) bool
+
+// PostCheckFunc is an optional filter executed after CheckTx and rejects
+// transaction if false is returned.
```

Comment:
Perhaps add that an example usage would be to ensure a tx doesn't have too much gas to ever fit into a block?